### PR TITLE
Enable collecting in memory config locally by default

### PR DESCRIPTION
**Pull request by flands**
Configured main.go to enable in memory config gathering locally. In
addition, updated support bundle to capture this information.

### CODE REVIEW

File: cmd/otelcol/main.go [hunk 1]
```diff
@@ -35,12 +35,13 @@ import (
...
 const (
-	ballastEnvVarName     = "SPLUNK_BALLAST_SIZE_MIB"
-	configEnvVarName      = "SPLUNK_CONFIG"
-	memLimitMiBEnvVarName = "SPLUNK_MEMORY_LIMIT_MIB"
-	memTotalEnvVarName    = "SPLUNK_MEMORY_TOTAL_MIB"
-	realmEnvVarName       = "SPLUNK_REALM"
-	tokenEnvVarName       = "SPLUNK_ACCESS_TOKEN"
+	ballastEnvVarName          = "SPLUNK_BALLAST_SIZE_MIB"
+	configEnvVarName           = "SPLUNK_CONFIG"
+	configServerEnabledEnvVar = "SPLUNK_DEBUG_CONFIG_SERVER"
```
Comment (pjanotti) [inline cmd/otelcol/main.go:40]: nit: spacing 
```suggestion
	configServerEnabledEnvVar  = "SPLUNK_DEBUG_CONFIG_SERVER"
```

File: cmd/otelcol/main.go [hunk 2]
```diff
@@ -61,6 +62,10 @@ func main() {
 		checkRuntimeParams()
 	}
 
+	// Allow dumping configuration locally by default
+	// Used by support bundle script
+	os.Setenv(configServerEnabledEnvVar, "true")
```
Comment (pjanotti) [inline cmd/otelcol/main.go:67]: fyi: if the port is not available it would fail to start - I don't recall the exact message in this case.

**Review by pjanotti — APPROVED** [review]
LGTM, a minor formatting nit + and FYI.

File: cmd/otelcol/main.go (hunk 2 above)
Comment (flands) [reply cmd/otelcol/main.go:67]: bind: port already in use -- already hit it :)

**Review by rmfitzpatrick — APPROVED** [review]

**Pull request merged**